synapse/storage/schema/delta/56/stats_separated1.sql

```diff
@@ -31,3 +31,118 @@ DELETE FROM background_updates WHERE update_name IN (
     'populate_stats_process_rooms',
     'populate_stats_cleanup'
 );
+
+----- Create tables for our version of room stats.
+
+-- single-row table to track position of incremental updates
+CREATE TABLE IF NOT EXISTS stats_incremental_position (
+    -- the stream_id of the last-processed state delta
+    state_delta_stream_id BIGINT,
+
+    -- the stream_ordering of the last-processed backfilled event
+    -- (this is negative)
+    total_events_min_stream_ordering BIGINT,
+
+    -- the stream_ordering of the last-processed normally-created event
+    -- (this is positive)
+    total_events_max_stream_ordering BIGINT,
+
+    -- If true, this represents the contract agreed upon by the background
+    -- population processor.
+    -- If false, this is suitable for use by the delta/incremental processor.
+    is_background_contract BOOLEAN NOT NULL PRIMARY KEY
+);
+
+-- insert a null row and make sure it is the only one.
+DELETE FROM stats_incremental_position;
+INSERT INTO stats_incremental_position (
+    state_delta_stream_id,
+    total_events_min_stream_ordering,
+    total_events_max_stream_ordering,
+    is_background_contract
+) VALUES (NULL, NULL, NULL, (0 = 1)), (NULL, NULL, NULL, (1 = 1));
+
+-- represents PRESENT room statistics for a room
+CREATE TABLE IF NOT EXISTS room_stats_current (
+    room_id TEXT NOT NULL PRIMARY KEY,
+
+    -- These starts cover the time from start_ts...end_ts (in seconds).
+    -- Note that end_ts is quantised, and start_ts usually so.
```

Comment:
Why wouldn't `start_ts` be quantised?

Reply:
`start_ts` isn't quantised if it comes about from the creation of a new room/user.

It *could* be quantised but I wasn't sure how much sense it makes either way; I was on the fence about it and decided to go with this.

Reply:
> `start_ts` isn't quantised if it comes about from the creation of a new room/user.
> 
> It _could_ be quantised but I wasn't sure how much sense it makes either way; I was on the fence about it and decided to go with this.

(If you imagine drawing a histogram of the data, which is the form that these tables take in my mind, then it seems like it'd be wrong to draw the start of this bar BEFORE the start of the room's/user's existence. Also consider if we use the data to calculate a rate, e.g. <x> events / <time period> – it wouldn't be quite right to artificially extend the time slice to include time that wasn't actually tracked).

Does this make sense? If so, should I also put this explanation in the schema?